File: Core.DomainModel/ItSystem/DataRow.cs
using System;

namespace Core.DomainModel.ItSystem
{
    /// <summary>
    /// Represents that an interface exposes some data.
    /// The interface will have a datarow for each piece of data that the
    /// interface exposes.
    /// </summary>
    public class DataRow : Entity, ISystemModule, IHasUuid
    {
        public DataRow()
        {
            Uuid = Guid.NewGuid();
        }

        public int ItInterfaceId { get; set; }
        /// <summary>
        /// The interface which exposes the data
        /// </summary>
        public virtual ItInterface ItInterface { get; set; }

        public int? DataTypeId { get; set; }
        /// <summary>
        /// The type of the data
        /// </summary>
        public virtual DataType DataType { get; set; }

        /// <summary>
        /// Description/name of the data
        /// </summary>
        public string Data { get; set; }

        public Guid Uuid { get; set; }
    }
}

File: Presentation.Web/app/models/it-system/data-row.ts
module Kitos.Models.ItSystem {
    export interface IDataRow extends IEntity {
        ItInterfaceId: number;
        /** The interface which exposes the data */
        ItInterface: IItInterface;
        DataTypeId: number;
        /** The type of the data */
        DataType: Models.OData.Generic.IOptionDTO<IDataRow>;
        /** Description/name of the data */
        Data: string;
    }
}
